src/csync/csync_reconcile.cpp

```diff
@@ -187,20 +168,18 @@ static int _csync_merge_algorithm_visitor(void *obj, void *data) {
             }
 
             if( tmp ) {
-                len = strlen( tmp->path );
-                if( len > 0 ) {
-                    h = c_jhash64((uint8_t *) tmp->path.constData(), len, 0);
+                if( !tmp->path.isEmpty() ) {
                     /* First, check that the file is NOT in our tree (another file with the same name was added) */
-                    node = c_rbtree_find(ctx->current == REMOTE_REPLICA ? ctx->remote.tree : ctx->local.tree, &h);
-                    if (node) {
+                    csync_s::FileMap *our_tree = ctx->current == REMOTE_REPLICA ? &ctx->remote.files : &ctx->local.files;
+                    if (our_tree->find(tmp->path) != our_tree->end()) {
                         CSYNC_LOG(CSYNC_LOG_PRIORITY_TRACE, "Origin found in our tree : %s", tmp->path.constData());
                     } else {
                         /* Find the temporar file in the other tree. */
-                        node = c_rbtree_find(tree, &h);
-                        CSYNC_LOG(CSYNC_LOG_PRIORITY_TRACE, "PHash of temporary opposite (%s): %" PRIu64 " %s",
-                                tmp->path.constData() , h, node ? "found": "not found" );
-                        if (node) {
-                            other = (csync_file_stat_t*)node->data;
+                        other_file_it = other_tree->find(tmp->path);
+                        CSYNC_LOG(CSYNC_LOG_PRIORITY_TRACE, "Temporary opposite (%s) %s",
+                                tmp->path.constData() , other_file_it != other_tree->end() ? "found": "not found" );
+                        if (other_file_it != other_tree->end()) {
+                            other = other_file_it->second.get();
```

Comment:
It may be convenient to have a function
```
find(tree, path) -> csync_file_stat_t *
```
Then we could skip dealing with iterators in most places.

Reply:
Done. Made FileMap a subclass and implemented it as `csync_file_stat_t *findFile(const QByteArray &key);`